### DISCUSSION TITLE
MediaElement stops playing after Android screen locks

### DISCUSSION THREAD

**Discussion by sbalkum** [CommunityToolkit/Maui#2121] [Q&A] [answered]
I'm hoping someone with MediaElement experience can provide some guidance.

I'm having trouble playing internet hosted mp3 files on an Android device after the screen locks. After a few songs, MediaElement fails to play the next song, apparently with an UnknownHostException from ExoPlayer.  This appears similar to a previous [bug](https://github.com/CommunityToolkit/Maui/issues/1362), but that bug was supposedly fixed with MediaElement 4.0.

I have placed a barebones demonstration project on [github](https://github.com/sbalkum/PlayMp3sForever). I made all of the platform specific additions mentioned by MediaElement's [documentation](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/views/mediaelement?tabs=android). This little app works perfectly on iOS.

The code is purposefully simple. Pick a URL at random and set the MediaElement's Source. When that media ends, do it again. Relevant lines:

MainPage.xaml:
```
...
            <toolkit:MediaElement x:Name="_player" ShouldShowPlaybackControls="True" ShouldAutoPlay="True" />
...
```

Mainpage.xml.cs:
```
...
        public MainPage()
        {
            InitializeComponent();

            _player.MediaEnded += (s, e) =>
            {
                SetMedia();
            };
...
        }
...
        protected override void OnAppearing()
        {
            base.OnAppearing();
...
            SetMedia();
        }

...
        public void SetMedia()
        {
            var i = Random.Shared.Next(0, _mp3Urls.Count);
            var url = _mp3Urls[i];
  
            Dispatcher.Dispatch(() =>
            {
                _player.Source = MediaSource.FromUri(url);
                _player.MetadataTitle = $"Title {DateTime.Now.Ticks}";
                _player.MetadataArtist = $"Artist {DateTime.Now.Ticks}";
                _player.MetadataArtworkUrl = "https://cdn.pixabay.com/photo/2016/03/29/10/38/piano-1287912_1280.png";

                _player.Play();
            });
        }
...
```

Thanks in advance for any help.

**Comment by ne0rrmatrix** [comment]
1. Can u update the URL so it is always the same media? 
2. If you are selecting random media you might be having issues with the player trying to play an incompatible source. 
3. I would try the same URL, with the same media. 
4. If that works then try random URL's till one breaks. 
5. If it does not try different types of media till one breaks. 
6. If it breaks on the same media being repeated from a single URL that is important to know. 

Please test further and get back to me. I will be happy to make any changes to media element once you determine what the exact issue is. 

I am in the process of adding Appium UI testing to the Community toolkit and in the future this should help developers automate all of this.

**Reply by sbalkum** [reply]
Sorry. My code snippet was too brief. The "random URLs" are actually just a hardcoded list of 5. My original test was using the same URL over and over again and it stopped playing after a few songs after screen as well. I expanded to 5 to make sure it wasn't related to playing the same mp3 repeatedly. If I prevent the screen lock, it plays repeated with no issues.

I have updated the code to only use a single URL and it failed a few songs after the screen locked.

I've been beating on this issue off and on for several weeks while finishing the full media player app. (That app is now complete and works great on iOS.)

Thanks for the assistance!

**Reply by ne0rrmatrix** [reply, marked as answer]
Can you test against the current main branch? The fix has not made it into public nuget yet. The last release was early July for media element.

**Reply by sbalkum** [reply]
Excellent! Yes, the main branch works. I let my test play for 20 repeats with the screen locked and then tried in my media player app and it played an entire 1 hour playlist. I'm guessing this is bugfix #2052. Do you have an estimated date when the nuget package will be updated?

**Comment by ne0rrmatrix** [comment]
Gerald is planning for today or tomorrow. Fingers crossed 🤞

**Reply by sbalkum** [reply]
That works for me. Thanks for the help!

**Comment by CliffAgius** [comment]
Closed as Answered.
